src/Application.cpp

```diff
@@ -0,0 +1,176 @@
+    // Couldn't set video mode.
+    assert(window);
+
+    // Get gl context and set it to the current context for this window.
+    glContext = SDL_GL_CreateContext(window);
+    // Failed to create OpenGL context.
+    assert(glContext);
+
+    // Don't know if this is needed.
+    glewInit();
+
+    // Enable v-sync.
+    SDL_GL_SetSwapInterval(1);
+
+    // Workaround for AMD. Must not be removed.
+    if(!glBindFragDataLocation) {
+        glBindFragDataLocation = glBindFragDataLocationEXT;
+    }
+}
+
+/*
+ * Cleans up after SDL.
+ */
+void Application::terminateSDL() {
+    //Destroy window
+    SDL_DestroyWindow(window);
+
+    //Quit SDL subsystems
+    SDL_Quit();
+}
+
+/*
+ * Call from main. Is the main loop of this application.
+ */
+void Application::start() {
+    initSDL();
+
+    RenderView renderView;
+
+    bool quit = false;
+    auto startTime = std::chrono::system_clock::now();
+    float currentTime = 0.0f;
+    float previousTime = 0.0f;
+    float deltaTime = 0.0f;
+
+    TEMP_setTexture("../src/title.pixdata", renderView);
+
+    while(!quit) {
+        //update currentTime
+        std::chrono::duration<float> timeSinceStart = std::chrono::system_clock::now() - startTime;
+        previousTime = currentTime;
+        currentTime = timeSinceStart.count();
+        deltaTime = currentTime - previousTime;
+
+        // Resize window to the size of the render view if needed.
+        {
+            int w, h;
+            SDL_GetWindowSize(window, &w, &h);
+            if (w != renderView.getWidth() || h != renderView.getHeight()) {
+                SDL_SetWindowSize(window, renderView.getWidth(), renderView.getHeight());
+            }
+        }
+
+        // render to window
+        renderView.render();
+
+        // Swap front and back buffer. This frame will now been displayed.
+        SDL_GL_SwapWindow(window);
+
+        // Check if it is time to quit or not
+        {
+            SDL_Event event;
+            while (SDL_PollEvent(&event)) {
+                switch (event.type) {
+                    case SDL_QUIT:
+                        quit = true;
+                        break;
+                    case SDL_KEYDOWN:
+                        switch( event.key.keysym.sym ){
+                            case SDLK_1:
+                                renderView.setPalette(PALETTE_POCKET);
+                                break;
+                            case SDLK_2:
+                                renderView.setPalette(PALETTE_DMG);
+                                break;
+                            case SDLK_3:
+                                renderView.setPalette(PALETTE_DMG_SMOOTH);
+                                break;
+                            case SDLK_4:
+                                break;
+                            default:
+                                break;
+                        }
+                }
+            }
+        }
```

Comment:
Delegate this to its own function maybe? Like `handleEvents()`?

Reply:
Almost everything there is temporary. I guess what one could do is to delegate event handling to its own class or function where we can add controller mapping support etc. However, as I said last meeting: there are lots of things that need to be done in Application and RenderView, none of which should be prioritized right now. At least not until the emulator part is in a more complete state.